File: pygit/workspace.py
import os


class Workspace:

    def __init__(self, pathname):
        self.pathname = pathname

    def list_files(self):
        return [name for name in os.listdir() if name.endswith('.py')]

    def read_file(self, path):
        path = os.path.join(self.pathname, path)
        return open(path, 'rb').read()

File: jgit/com/git/Workspace.java
package com.git;

import java.io.File;
import java.io.IOException;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.nio.file.Files;
import java.util.Collections;
import java.util.List;
import java.util.stream.Stream;
import java.util.stream.Collectors;

class Workspace {
    private String pathname;

    public Workspace(String pathname) {
        this.pathname = pathname;
    }

    public List<String> listFiles() {
        try (Stream<Path> walk = Files.walk(Paths.get(pathname))) {
             return walk.map(x -> x.toString())
                        .filter(f -> f.endsWith(".java"))
                        .collect(Collectors.toList());
        } catch (IOException e) {
            e.printStackTrace();
        }
        return Collections.emptyList();
    }

    public byte[] readFile(String path) throws IOException {
        return Files.readAllBytes(new File(path).toPath());
    }
}
